packages/frontend/src/features/sqlRunner/components/Header/HeaderEdit.tsx

```diff
@@ -121,6 +130,20 @@ export const HeaderEdit: FC = () => {
         }
     }, [hasErrors, config, sql, dispatch, onSave]);
 
+    const handleGoBackToViewPage = useCallback(async () => {
+        await queryClient.resetQueries({
+            queryKey: [
+                'sqlRunner',
+                'savedSqlChart',
+                projectUuid,
+                savedSqlChart?.slug,
+            ],
+        });
+        history.push(
+            `/projects/${projectUuid}/sql-runner/${savedSqlChart?.slug}`,
+        );
+    }, [queryClient, history, savedSqlChart, projectUuid]);
```

Comment:
ensures that if the user goes back, that we reset the queries of the savedSqlChart